galaxy_ng/app/settings.py

```diff
@@ -190,3 +191,96 @@
+    token_url_str = "{keycloak}/auth/realms/{realm}/protocol/openid-connect/token/"
+    SOCIAL_AUTH_KEYCLOAK_ACCESS_TOKEN_URL = token_url_str.format(keycloak=KEYCLOAK_URL,
+                                                                 realm=KEYCLOAK_REALM)
+
+    SOCIAL_AUTH_LOGIN_REDIRECT_URL = os.getenv('SOCIAL_AUTH_LOGIN_REDIRECT_URL', default='/ui/')
+    SOCIAL_AUTH_POSTGRES_JSONFIELD = True
+    SOCIAL_AUTH_JSONFIELD_CUSTOM = 'django.contrib.postgres.fields.JSONField'
+    SOCIAL_AUTH_URL_NAMESPACE = 'social'
+    SOCIAL_AUTH_KEYCLOAK_EXTRA_DATA = [('refresh_token', 'refresh_token'),
+                                       (KEYCLOAK_ROLE_TOKEN_CLAIM, KEYCLOAK_ROLE_TOKEN_CLAIM), ]
+
+    SOCIAL_AUTH_PIPELINE = (
+        'social_core.pipeline.social_auth.social_details',
+        'social_core.pipeline.social_auth.social_uid',
+        'social_core.pipeline.social_auth.social_user',
+        'social_core.pipeline.user.get_username',
+        'social_core.pipeline.social_auth.associate_by_email',
+        'social_core.pipeline.user.create_user',
+        'social_core.pipeline.social_auth.associate_user',
+        'social_core.pipeline.social_auth.load_extra_data',
+        'social_core.pipeline.user.user_details',
+        'social_core.pipeline.debug.debug',
+        'galaxy_ng.app.pipelines.user_role',
+        'galaxy_ng.app.pipelines.user_group',
+    )
+
+    # Set external authentication feature flag
+    GALAXY_FEATURE_FLAGS["external_authentication"] = True
+
+    # Add to installed apps
+    INSTALLED_APPS.append('social_django')
+
+    # Add to authentication backends
+    AUTHENTICATION_BACKENDS = [
+        # BEGIN: Pulp standard authentication backends
+        "django.contrib.auth.backends.ModelBackend",
+        "guardian.backends.ObjectPermissionBackend",
+        # END: Pulp standard authentication backends
+        'social_core.backends.keycloak.KeycloakOAuth2',
+    ]
+
+    # Build paths inside the project like this: BASE_DIR / ...
+    BASE_DIR = Path(__file__).absolute().parent
+
+    # Add to templates
+    TEMPLATES = [
+        {
+            "BACKEND": "django.template.backends.django.DjangoTemplates",
+            "DIRS": [BASE_DIR / "templates"],
+            "APP_DIRS": True,
+            "OPTIONS": {
+                "context_processors": [
+                    # BEGIN: Pulp standard context processors
+                    "django.template.context_processors.debug",
+                    "django.template.context_processors.request",
+                    "django.contrib.auth.context_processors.auth",
+                    "django.contrib.messages.context_processors.messages",
+                    # END: Pulp standard context processors
+                    'social_django.context_processors.backends',
+                    'social_django.context_processors.login_redirect',
+                ],
+            },
+        },
+    ]
```

Comment:
However in this case Dynaconf does not have a way to override itens inside a dict inside a list, I am opening an issue requesting this feature https://github.com/rochacbruno/dynaconf/issues/299#issuecomment-900616706

Reply:
👍

Reply:
I'd prefer it if we stick to using dynaconf for getting these settings from environment variables instead of using `os.getenv`. I think environment variables that start with `PULP_` get automatically picked up and added to settings, is that right @rochacbruno